operator/internal/manifests/var.go

```diff
@@ -619,3 +621,26 @@ func lokiReadinessProbe() *corev1.Probe {
 		FailureThreshold:    3,
 	}
 }
+
+// isZoneawareComponent returns true if the Component pod is either in the read/write path or gateway.
+func CheckZoneawareComponent(pod *corev1.Pod) bool {
+	podLabels := pod.Labels
+
+	if podLabels[kubernetesComponentLabel] == LabelDistributorComponent || podLabels[kubernetesComponentLabel] == LabelIndexGatewayComponent ||
+		podLabels[kubernetesComponentLabel] == LabelIngesterComponent || podLabels[kubernetesComponentLabel] == LabelQuerierComponent ||
+		podLabels[kubernetesComponentLabel] == LabelQueryFrontendComponent {
+		return true
+	}
```

Comment:
This is purely screaming for a switch statement to improve readability.

Reply:
Done